### PR TITLE
Content Localization

### PR DESCRIPTION
**Pull request by Skrypt**
Initially from @jptissot I'm reopening this PR from a branch where I fixed some NRE issues. It will be easier to add commits on top of this branch instead of his personnal fork.

For previous comments see : https://github.com/OrchardCMS/OrchardCore/pull/3380

Make sure that @jptissot gets the credit for the work 😄

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\..\OrchardCore.Modules\OrchardCore.ContentLocalization\OrchardCore.ContentLocalization.csproj" />
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj]: None of these references should be in this group, move them with the other ones, ordered

Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj]: With privateassets = none?
And do I move also : <ProjectReference Include="..\..\OrchardCore\OrchardCore.Infrastructure\OrchardCore.Infrastructure.csproj" />

Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj]: Not Infrastructure, it's not a module, but it should probably already be referenced by something else and not necessary here

Comment (jtkech) [reply src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj]: > privateassets = none?

Yes for all modules and Infrastructure is certainly not needed.

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/SummaryAdminDriver.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            _contentDefinitionManager = contentDefinitionManager;
+        }
+
+        public override IDisplayResult Display(ContentItem model, IUpdateModel updater)
+        {
+            var localizationPart = model.As<LocalizationPart>();
+            if (localizationPart != null)
+            {
+                // This injects a button on the SummaryAdmin view for the Hackathon ContentType
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/SummaryAdminDriver.cs]: Remove this comment

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Localization PR
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/README.md]: Remove this file or actually write docs

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Views/LocalizationPart.Edit.cshtml [hunk 4]
```diff
@@ -0,0 +1,91 @@
+@using System.Globalization
+@model LocalizationPartViewModel
+
+<style>
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Views/LocalizationPart.Edit.cshtml]: Move this in another file (I have done this, will link a branch soon.

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/LocalizationPartDriver.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            var deletedCultureTranslations = alreadyTranslated.Select(ci =>
+              {
+                  var culture = ci.As<LocalizationPart>()?.Culture;
+                  if (currentCultures.Any(c=>c.ContentItem?.ContentItemId == ci.ContentItemId))
+                  {
+                      return null;
+                  }
+                  return new LocalizationLinksViewModel()
+                  {
+                      IsDeleted = true,
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Drivers/LocalizationPartDriver.cs]: Not sure this is actually required. Do we want to display the fact that a Localization was deleted ?

**Comment by jptissot** [comment]
@Skrypt check this branch for small tweaks. https://github.com/jptissot/OrchardCore/tree/skrypt/content-localization

**Pull request closed without merging**